stmhal/uart.c

```diff
@@ -115,7 +115,7 @@ void uart_deinit(void) {
 STATIC bool uart_init2(pyb_uart_obj_t *uart_obj) {
     USART_TypeDef *UARTx;
     IRQn_Type irqn;
-    uint32_t GPIO_Pin, GPIO_Pin2;
+    uint32_t GPIO_Pin, GPIO_Pin2= NULL;
```

Comment:
Enter
arm-none-eabi-gcc -I../lib/netutils -I. -I.. -Ibuild-STM32F429DISC -Icmsis/inc -Icmsis/devinc -Ihal/f4/inc -Iusbdev/core/inc -Iusbdev/class/inc -I../lib/mp-readline -I../lib/netutils -I../lib/timeutils -Wall -Wpointer-arith -Werror -ansi -std=gnu99 -nostdlib  -DSTM32F429xx -mthumb -mabi=aapcs-linux -mfpu=fpv4-sp-d16 -mfloat-abi=hard -fsingle-precision-constant -Wdouble-promotion -mtune=cortex-m4 -mcpu=cortex-m4 -DMCU_SERIES_F4 -O0 -Iboards/STM32F429DISC -DSTM32_HAL_H='<stm32f4xx_hal.h>' -g -DPENDSV_DEBUG -c -MD -o build-STM32F429DISC/uart.o uart.c
Gives me an error of uninitialized variable which is true if you check the code. I'll check why the makefile does not complain.